infra/src/main/kotlin/fi/fta/geoviite/api/frameconverter/v1/FrameConverterServiceV1.kt

```diff
@@ -0,0 +1,343 @@
+package fi.fta.geoviite.api.frameconverter.v1
+
+import fi.fta.geoviite.api.frameconverter.geojson.GeoJsonFeature
+import fi.fta.geoviite.api.frameconverter.geojson.GeoJsonGeometryPoint
+import fi.fta.geoviite.infra.aspects.GeoviiteService
+import fi.fta.geoviite.infra.common.AlignmentName
+import fi.fta.geoviite.infra.common.LayoutContext
+import fi.fta.geoviite.infra.common.TrackNumber
+import fi.fta.geoviite.infra.error.InputValidationException
+import fi.fta.geoviite.infra.geocoding.AddressAndM
+import fi.fta.geoviite.infra.geocoding.GeocodingService
+import fi.fta.geoviite.infra.localization.LocalizationLanguage
+import fi.fta.geoviite.infra.localization.LocalizationParams
+import fi.fta.geoviite.infra.localization.LocalizationService
+import fi.fta.geoviite.infra.math.Point
+import fi.fta.geoviite.infra.math.boundingBoxAroundPoint
+import fi.fta.geoviite.infra.math.lineLength
+import fi.fta.geoviite.infra.publication.LayoutValidationIssueType
+import fi.fta.geoviite.infra.publication.LayoutValidationIssueType.ERROR
+import fi.fta.geoviite.infra.publication.validateWithParams
+import fi.fta.geoviite.infra.tracklayout.AlignmentPoint
+import fi.fta.geoviite.infra.tracklayout.LayoutAlignment
+import fi.fta.geoviite.infra.tracklayout.LayoutTrackNumberService
+import fi.fta.geoviite.infra.tracklayout.LocationTrack
+import fi.fta.geoviite.infra.tracklayout.LocationTrackService
+import fi.fta.geoviite.infra.tracklayout.LocationTrackType
+import org.springframework.beans.factory.annotation.Autowired
+import java.math.BigDecimal
+import java.math.RoundingMode
+
+private data class NearestLocationTrackPointMatch(
+    val locationTrack: LocationTrack,
+    val alignment: LayoutAlignment,
+    val closestPointOnTrack: AlignmentPoint,
+    val distanceToClosestPoint: Double,
+)
+
+private val allowedSearchRadiusRange = 1.0..1000.0
+
+@GeoviiteService
+class FrameConverterServiceV1 @Autowired constructor(
+    private val trackNumberService: LayoutTrackNumberService,
+    private val geocodingService: GeocodingService,
+    private val locationTrackService: LocationTrackService,
+    localizationService: LocalizationService,
+) {
+    val translation = localizationService.getLocalization(LocalizationLanguage.FI)
+
+    fun coordinateToTrackAddress(
+        layoutContext: LayoutContext,
+        request: ValidCoordinateToTrackMeterRequestV1,
```

Comment:
Funkkari edellyttää validoidun rakenteen passaamista. Näin funkkarin vaatimukset päätyvät jo tyyppitiedossa kutsujalle, eikä funkkarissa tarvitse tehdä erikseen enää validointia ja muuta varsinaiselle kutsulle (null-checkit, toString-konversiot meidän sisäisten tyyppien vertailuissa jne)